haproxy_discovery.sh

```diff
@@ -11,7 +11,24 @@
 #  global
 #  stats socket /run/haproxy/info.sock  mode 666 level user
 
-HAPROXY_SOCK="/var/run/haproxy/info.sock"
+SCRIPT_DIR=`dirname $0`
+CONF_FILE="${SCRIPT_DIR}/haproxy_zbx.conf"
+CONF_MAP="
+ha_sock=/var/run/haproxy/info.sock
+debug=0
+discovery_log_file=/var/tmp/haproxy_discovery.log
+"
+get_conf() {
+    local conf_val=`grep $1= ${CONF_FILE} | xargs -d "=" | grep $1 |awk '{print $2}'`
+    if [ $conf_val ]; then
+        echo $conf_val
+    else
+        echo "`echo -e "$CONF_MAP" | grep $1=  | xargs -d "=" | grep $1 |awk '{print $2}'`"
+    fi
+}
```

Comment:
I like the idea of conf file.. 
Though, wouldn't be easier to just source the file?
What's the advantage of using `get_conf` over sourcing the file?
I mean, I see how it uses `CONF_MAP` to lookup a conf_val if it's not in config file, but setting some defaults and then overriding them by sourcing the file is way simpler.
```
ha_sock=/var/run/haproxy/info.sock
debug=0
discovery_log_file=/var/tmp/haproxy_discovery.log

if [[ -r $CONF_FILE ]]; then
  source ${CONF_FILE}
fi
```

Reply:
That's a good idea, I'll work on it tomorrow +1

Reply:
changed on both scripts to use ENV vars